### ISSUE TITLE
Allow accessing state's internal pool

### ISSUE THREAD

**Issue by arcstur** [digital-society-coop/axum-sqlx-tx#41]
Sometimes I need the pool, to construct other stuff in the application. It would be useful to access the state's internal pool, so that I don't need to have two of the same pool in my app's state. What do you think?

What I mean is that I don't want to have

```rust
#[derive(Clone, FromRef)]
pub struct AppState {
    pub pool: PgPool,
    pub tx_state: axum_sqlx_tx::State<sqlx::Postgres>,
}
```

And have only the `tx_state`, since it already holds a pool.

**Comment by connec** [comment]
It would be possible to `impl FromRef<axum_sqlx_tx::State<DB>> for sqlx::Pool<DB>`, meaning a `State<PgPool>` could be extracted from an `axum_sqlx_tx::State<sqlx::Postgres>`.

Would that work for your use-case?

If you already needed a separate state struct, you'd have to proxy the `FromRef` implementation manually.

**Comment by arcstur** [comment]
Yeah, that would be awesome! Then, in my state struct, I would do `fn pool(&self) { PgPool::from_ref(&self.tx_state) }`

**Comment by connec** [comment]
Fixed by #42.
